### ISSUE TITLE
 Undefined reference to `__f03_unl_poly_0____td_'

### ISSUE THREAD

**Issue by nncarlson** [flang-compiler/flang#307]
The current master branch version of flang is giving me this link error for the example code that follows. I'm able to compile and link other programs with flang just fine, so I'm inclined to think this is due to a compiler error rather than a problem with my environment.
>>>
    $ flang flang-20171122a.f90 
    /tmp/flang-20171122a-6ad8d1.o: In function `MAIN_':
    flang-20171122a.f90:6: undefined reference to `__f03_unl_poly_0____td_'
    clang-5.0: error: linker command failed with exit code 1 (use -v to see invocation)
>>>
```fortran
class(*), allocatable :: val(:)
call get_value (val)
select type (val)
type is (character(*))
  print '(2i2,1x,2a)', len(val), size(val), val ! expect 3 2 foobar
end select
contains
  subroutine get_value (value)
    class(*), allocatable, intent(out) :: value(:)
    allocate(value, source=['foo','bar'])
  end subroutine
end
```

**Comment by nncarlson** [comment]
Still present in 1efaacf

**Comment by nncarlson** [comment]
Fixed in 543785e7
